### PR TITLE
refactor: control buttons

### PR DESCRIPTION
**Pull request by lauty95**
WIP
An idea of use the ControlButtons component based on issue #215 

```vue
<template>
    <ControlButtons :buttons="controlButtons" />
</template>
<script>
const controlButtons = computed(() => {
	return [
		{
			label: 'Save',
			action: create,
			disabled: listItems.value.length === 0,
		},
		{
			label: 'Submit',
			action: () => store.submit<StockEntry>('Stock Entry', stockEntryId.value),
			disabled: stockEntryId.value === '',
		},
		{
			label: 'Cancel',
			action: () => store.cancel<StockEntry>('Stock Entry', stockEntryId.value),
			disabled: stockEntryId.value === '',
		},
	]
})
</script>
```

i have to work on styles, but based on the length of array the buttons should fit in its place

Captura de pantalla 2024-11-13 a la(s) 11 41 29 a  m
Captura de pantalla 2024-11-13 a la(s) 11 41 57 a  m
Captura de pantalla 2024-11-13 a la(s) 11 44 07 a  m

### CODE REVIEW

File: beam/www/beam/components/ControlButtons.vue [hunk 1]
```diff
@@ -1,78 +1,40 @@
...
+	justify-content: flex-end;
+}
+
+.button-2 {
+	justify-content: space-between;
+}
+
+.button-3 {
+	justify-content: space-between;
+}
```
Comment (Alchez) [inline beam/www/beam/components/ControlButtons.vue:39]: How would this work for more than 3 buttons? Should these be made dynamic using parent flex justify values?

Comment (lauty95) [reply beam/www/beam/components/ControlButtons.vue:39]: Ignore my custom classes haha, just did that way to make some photos.

But I'll think about it for any number of buttons

Comment (agritheory) [reply beam/www/beam/components/ControlButtons.vue:39]: @lauty95 I think we don't need to support more than three. I'd be OK with stacking if we have three or four.

**Comment by agritheory** [comment]
@lauty95 Can you add a "hidden" API in addition to "disabled"? I can think of a use case for each in the workflow we've put together already. Hidden should evaluate first and override disabled.

**Comment by agritheory** [comment]
This was super easy: 
```
		<BeamBtn
			v-for="(button, index) in buttons"
			:key="index"
			@click="button.action"
			:disabled="button.disabled"
			v-show="!button.hidden"
		>
		```

**Review by agritheory — APPROVED** [review]
Great work. Merging even though it's in WIP

**Pull request merged**